### PR TITLE
[Sprint3] Task 6807 fix register with phone number

### PR DESCRIPTION
**Pull request by manhpv01**
link : https://edu-redmine.framgia.vn/issues/6807

### CODE REVIEW

File: app/src/main/java/com/framgia/oleo/data/source/UserDataSource.kt [hunk 1]
```diff
@@ -16,5 +19,13 @@ interface UserDataSource {
 
     interface Remote {
         fun pushUserLocation(id: String, place: Place)
+
+        fun getUserByPhoneNumber(phoneNumber: String, valueEventListener: ValueEventListener)
+
+        fun addNewUser(
```
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/oleo/data/source/UserDataSource.kt]:
```suggestion
        fun registerUser(
```

File: app/src/main/java/com/framgia/oleo/data/source/remote/api/user/UserRemoteDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import com.framgia.oleo.data.source.UserDataSource
+import com.framgia.oleo.data.source.model.Place
+import com.framgia.oleo.data.source.model.User
+import com.framgia.oleo.utils.Constant
+import com.google.android.gms.tasks.OnCompleteListener
+import com.google.android.gms.tasks.OnFailureListener
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+
+class UserRemoteDataSource : UserDataSource.Remote {
```
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/oleo/data/source/remote/api/user/UserRemoteDataSource.kt]: sao lại có 2 class UserRemoteDataSource vậy e

File: app/src/main/java/com/framgia/oleo/screen/signup/SignUpFragment.kt [hunk 3]
```diff
@@ -115,34 +116,50 @@ class SignUpFragment : DialogFragment(), Injectable, View.OnClickListener {
...
-                        progressDialog.dismiss()
-                        if (result.isSuccessful) {
-                            onResultWhenLoginSuccess!!.invoke(textInputPhoneNumber.text.toString())
-                            dismiss()
-                        } else textMessageSignUp.text = getString(R.string.sign_up_fail)
+        progressDialog.show()
+        viewModel.getUserByPhoneNumber(textInputPhoneNumber.text.toString(), object : ValueEventListener {
+            override fun onDataChange(dataSnapShot: DataSnapshot) {
+                if (isSignUp) {
+                    return
```
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/oleo/screen/signup/SignUpFragment.kt]: if (isSignUp) return

File: app/src/main/java/com/framgia/oleo/screen/signup/SignUpViewModel.kt [hunk 4]
```diff
@@ -3,9 +3,25 @@ package com.framgia.oleo.screen.signup
...
+import com.google.firebase.database.ValueEventListener
 import javax.inject.Inject
 
-class SignUpViewModel @Inject constructor() : BaseViewModel() {
+class SignUpViewModel @Inject constructor(private val userRepository: UserRepository) : BaseViewModel() {
+    fun getUserByPhoneNumber(phoneNumber: String, valueEventListener: ValueEventListener) {
+        userRepository.getUserByPhoneNumber(phoneNumber, valueEventListener)
+    }
+
+    fun addNewUser(
```
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/oleo/screen/signup/SignUpViewModel.kt]:
```suggestion
    fun registerUser(
```

File: app/src/main/java/com/framgia/oleo/data/source/remote/api/user/UserRemoteDataSource.kt (hunk 2 above)
Comment (manhphan34) [reply app/src/main/java/com/framgia/oleo/data/source/remote/api/user/UserRemoteDataSource.kt]: cai nay e quen chua xoa bot 1 cai

**Pull request merged**